### PR TITLE
Single cluster deployment

### PR DESCRIPTION
**Pull request by kim-cc**
### Description
Simplifies the Encryptonize deployment to only consist of one cluster.
This means that only Encryptonize is available externally, the rest of the components aren't, instead you would have to port-forward to access them directly.

### Relevant Issues/PRs
Issue #270 

### Type(s) of Change (Split the PR if you check many)
- [ ] Added user feature
- [ ] Added technical feature
- [ ] Added tests
- [X] Refactor
- [ ] Bugfix
- [ ] Updated documentation
- [X] Kubernetes changes
- [ ] CI/CD workflow changes

### Testing Checklist
- [ ] I have added/updated unit tests for functional changes.
- [ ] I have added/updated integration tests for changes that affect dependent systems.
- [ ] I have added/updated end-to-end tests for feature changes.

### General Checklist
- [X] I have pulled in the latest changes from master.
- [X] I have run `make lint`.
- [X] I have successfully run `make tests`.
- [ ] I have updated relevant CI/CD workflows.
- [X] I have updated relevant Kubernetes manifests/scripts.
- [X] I have updated/added relevant (internal and external) documentation (readme, doc comments, etc).
- [ ] I have updated the dependency map to reflect my changes.
- [ ] I have added the license to new source code files.
- [X] I have spent some time looking over the full diff before creating this PR.

### CODE REVIEW

File: kubernetes/scripts/bootstrap-encryptonize.sh [hunk 1]
```diff
@@ -24,16 +24,14 @@ OUT_DIR="${ROOT_DIR}/encryptonize/bootstrap/config"
 
 mkdir -p "${OUT_DIR}"
 
-# Get object cluster secrets
-gcloud container clusters get-credentials $OBJ_CLUSTER --zone $ZONE --project $PROJECT
+gcloud container clusters get-credentials $ENC_CLUSTER --zone $ZONE --project $PROJECT
```
[the remark this answers is not in the document]
Comment (kim-cc) [reply kubernetes/scripts/bootstrap-encryptonize.sh]: Fixed

File: kubernetes/scripts/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+set -euo pipefail
+
+ROOT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && cd .. && pwd )"
+
+${ROOT_DIR}/scripts/bootstrap-auth.sh
+${ROOT_DIR}/scripts/deploy-auth.sh
+${ROOT_DIR}/scripts/bootstrap-object.sh
+${ROOT_DIR}/scripts/deploy-object.sh
+${ROOT_DIR}/scripts/bootstrap-encryptonize.sh
+${ROOT_DIR}/scripts/deploy-encryptonize.sh
```
[the remark this answers is not in the document]
Comment (kim-cc) [reply kubernetes/scripts/deploy.sh]: Fixed

File: kubernetes/makefile [hunk 3]
```diff
@@ -83,9 +83,15 @@ bootstrap-logging: $(templated_files) ## bootstrap log collection
 
 .PHONY: deploy-logging
 deploy-logging: $(templated_files) ## Deploy log collection
+	$(MAKE) bootstrap-logging && \
```
[the remark this answers is not in the document]
Comment (kim-cc) [reply kubernetes/makefile]: Fixed

File: kubernetes/README.md [hunk 4]
```diff
@@ -12,9 +12,7 @@ straight forward, however, to adapt these scripts for other providers or self-ho
 
 Deploying Encryptonize&reg; consists of four main steps:
 1. [Initial Setup](#setup)
-2. [Deploying an Auth storage cluster](#auth-storage-deployment)
-3. [Deploying an Object storage cluster](#object-storage-deployment)
-4. [Deploying the Encryption Service](#encryption-service-deployment)
+2. [Deploying an Auth storage cluster](#deployment)
```
[the remark this answers is not in the document]
Comment (kim-cc) [reply kubernetes/README.md]: Fixed

[the remark this answers is not in the document]
Comment (kim-cc) [reply kubernetes/README.md]: Fixed

File: kubernetes/makefile [hunk 5]
```diff
@@ -83,9 +83,15 @@ bootstrap-logging: $(templated_files) ## bootstrap log collection
...
 	source $(provider_env) && \
 	./scripts/deploy-logging.sh
 
+.PHONY: deploy
+deploy: $(templated_files) $(secret_files) ## Deploy
```
[the remark this answers is not in the document]
Comment (kim-cc) [reply kubernetes/makefile]: Fixed

File: kubernetes/scripts/bootstrap-encryptonize.sh [hunk 6]
```diff
@@ -24,16 +24,14 @@ OUT_DIR="${ROOT_DIR}/encryptonize/bootstrap/config"
...
 
+# Get object cluster secrets
 echo -e "${BLUE_ON}[+] Fetching object storage secrets${COLOR_OFF}"
 kubectl -n rook-ceph get secret bucket-claim -o jsonpath="{.data['AWS_ACCESS_KEY_ID']}" | base64 -d > "${OUT_DIR}/object_storage_id"
```
[the remark this answers is not in the document]
Comment (kim-cc) [reply kubernetes/scripts/bootstrap-encryptonize.sh:32]: Fixed

**Review by philip-cc — COMMENTED** [review]
Looks good, a few minor things left.

File: kubernetes/makefile [hunk 7]
```diff
@@ -76,6 +86,11 @@ deploy-encryptonize: $(templated_files) ## Deploy encryptonize cluster
 	source $(provider_env) && \
 	./scripts/deploy-encryptonize.sh
 
+.PHONY: object
+object: ## Bootstrap and deploy encryptonize
```
[the remark this answers is not in the document]
Comment (kim-cc) [reply kubernetes/makefile:90]: Fixed

File: kubernetes/makefile [hunk 8]
```diff
@@ -86,6 +101,20 @@ deploy-logging: $(templated_files) ## Deploy log collection
...
+	$(MAKE) deploy-logging
+
+.PHONY: deploy
+deploy: ## Deploy full Encryptonize setup
+	$(MAKE) bootstrap-auth
+	$(MAKE) deploy-auth
+	$(MAKE) bootstrap-object
+	$(MAKE) deploy-object
+	$(MAKE) bootstrap-encryptonize
+	$(MAKE) deploy-encryptonize
```
[the remark this answers is not in the document]
Comment (kim-cc) [reply kubernetes/makefile:116]: Ahh yes of course :) Fixed

File: .github/workflows/encryption-service-deploy.yml [hunk 9]
```diff
@@ -64,6 +64,7 @@ jobs:
       shell: 'script -e -c "bash {0}"' # Forces a TTY so we can do "kubectl exec.."
       run: |
         make gcp-login
+        make cluster-credentials
```
[the remark this answers is not in the document]
Comment (kim-cc) [reply .github/workflows/encryption-service-deploy.yml:67]: Missed that, it has been fixed

**Review by philip-cc — APPROVED** [review]

**Pull request merged**